src/components/pickers/BasePicker.tsx

```diff
@@ -38,6 +39,8 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
 
   protected suggestionStore: SuggestionsController<T>;
   protected SuggestionOfProperType = Suggestions as new (props: ISuggestionsProps<T>) => Suggestions<T>;
+  protected loadingTimer: number;
+  protected currentPromise: PromiseLike<any>;
```

Comment:
Do these REALLY need to be protected? Please use protected very sparingly. It's like using public for everything. You should have unit tests for everything public or protected that validate they don't change.